packages/affiliates/test/e2e/datasets.js

```diff
@@ -231,9 +204,47 @@ describe("Datasets", function() {
       const cg = mocks.Coingecko(path);
       const gcprices = await cg.getHistoricContractPrices(params.collateralTokens[0]);
       assert(gcprices.length);
-      const sp = mocks.SynthPrices(path);
-      const spprices = await sp.getHistoricSynthPrices(params.empContracts[0]);
-      assert(spprices.length);
+    });
+  });
+  describe("Dataset and Mocks for Dapp Mining", function() {
+    let path;
+    after(function(done) {
+      fs.rmdir(basePath, { recursive: true }, done);
+    });
+    it("saves new set", async function() {
+      this.timeout(10000);
+      const config = {
+        empAddress: "0xE4256C47a3b27a969F25de8BEf44eCA5F2552bD5",
+        startTime: 1607558400000,
+        endTime: 1609113600000
+      };
+      const ds = Dataset(basePath, { queries });
+      path = await ds.saveDappMining("dapp-mining-test", config);
+    });
+    it("loads query mock", async function() {
+      this.timeout(100000);
+      const config = {
+        empAddress: "0xE4256C47a3b27a969F25de8BEf44eCA5F2552bD5",
+        startTime: 1607558400000,
+        endTime: 1609113600000
+      };
+      assert(path, "requires dataset path");
+      const query = mocks.Queries(path);
+      const blocks = await query.getBlocks(config.startTime, config.endTime);
+      assert(blocks.length);
+      const descBlocks = await query.getBlocksDescending(config.endTime, 2);
+      assert(descBlocks.length, 2);
```

Comment:
What is this assert doing? IIUC, this is asserting that the length is >= 0 and if it errors it prints an error message of 2?

Reply:
that was a typo, its supposed to be asserting that length is equal to 2!

Reply:
Might be worth doing a brief audit of these tests to ensure we're not doing this anywhere else!

@chrismaree another instance where it might be useful to have a lint rule around assert. Maybe the right way to structure it is a lint rule that requires the second argument to assert to always be a string literal.... although there may be cases where a string literal is being tested against, so that's not a perfect solution...

Reply:
Looked through the tests and did not see any other violations